### PR TITLE
dbld: add debian-bullseye

### PR DESCRIPTION
**Pull request by alltilla**
Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: dbld/images/debian-bullseye.dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ENV LANG C.UTF-8
+
+COPY images/fake-sudo.sh /usr/bin/sudo
+COPY images/entrypoint.sh /
+COPY . /dbld/
+
+RUN /dbld/builddeps install_dbld_dependencies
+RUN /dbld/builddeps install_apt_packages
+RUN /dbld/builddeps install_debian_build_deps
+RUN /dbld/builddeps install_criterion
```
Comment (gaborznagy) [inline dbld/images/debian-bullseye.dockerfile:22]: Just an idea for the future: we can start using `criterion` from system packages.
It's included in Debian Bullseye, Debian Testing, and in Ubuntu 21.04 and 21.10.
https://repology.org/project/criterion/versions

It's not used in syslog-ng's debian/control file yet
https://github.com/gcsideal/syslog-ng-debian/blob/debian/unstable/debian/control

**Review by OverOrion — DISMISSED** [review]
LGTM, I successfully built the image, then the deb packages on my machine.

**Review by gaborznagy — DISMISSED** [review]

File: dbld/images/debian-bullseye.dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (alltilla) [inline dbld/images/debian-bullseye.dockerfile:23]: `RUN /dbld/builddeps install_bison_from_source` is missing from here. See `Debian 11` at https://repology.org/project/bison/versions

Comment (gaborznagy) [reply dbld/images/debian-bullseye.dockerfile:23]: Bison is an optional dependency as we build from tarball, right?

Comment (alltilla) [reply dbld/images/debian-bullseye.dockerfile:23]: Yes, you are right, we do not install bison on any other of our "simple" builder platforms. The kira image has it, but it is justified.

File: dbld/images/debian-bullseye.dockerfile (hunk 1 above)
Comment (alltilla) [reply dbld/images/debian-bullseye.dockerfile:22]: Good idea! I will look into it. Please do not merge this PR until that.

File: .github/workflows/packages.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ jobs:
       matrix:
         distro: 
         - centos-7
-        - debian-buster
+        - debian-bullseye
```
Comment (szemere) [inline .github/workflows/packages.yml:39]: I just want to highlight this good decision. We do not simply add new platforms to the packages action. We use a carefully selected set of platforms for this.

File: dbld/images/debian-bullseye.dockerfile (hunk 1 above)
Comment (alltilla) [reply dbld/images/debian-bullseye.dockerfile]: I added it to `packaging/debian/control` :)

**Comment by kira-syslogng** [comment]
Build FAILURE

File: dbld/images/debian-bullseye.dockerfile (hunk 1 above)
Comment (gaborznagy) [reply dbld/images/debian-bullseye.dockerfile]: superb :)

**Review by gaborznagy — APPROVED** [review]

**Review by szemere — APPROVED** [review]

**Comment by szemere** [comment]
@kira-syslogng test this please;

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**